src/lib/orm/orm.go

```diff
@@ -84,27 +88,34 @@ func Clone(ctx context.Context) context.Context {
 // WithTransaction a decorator which make f run in transaction
 func WithTransaction(f func(ctx context.Context) error) func(ctx context.Context) error {
 	return func(ctx context.Context) error {
+		_, span := tracelib.StartTrace(ctx, tracerName, "start-transaction")
+		defer span.End()
 		o, err := FromContext(ctx)
 		if err != nil {
+			tracelib.RecordError(span, err, "get orm from ctx failed")
 			return err
 		}
 
 		tx := ormerTx{Ormer: o}
 		if err := tx.Begin(); err != nil {
+			tracelib.RecordError(span, err, "begin transaction failed")
 			log.Errorf("begin transaction failed: %v", err)
 			return err
 		}
 
 		if err := f(ctx); err != nil {
```

Comment:
The context created by the tracelib.StartTrace() should propagate to the children span